File: spring-data-jpa/src/main/java/study/spring/jpa/product/model/Product.java
package study.spring.jpa.product.model;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.extern.slf4j.Slf4j;

import javax.persistence.*;
import java.util.UUID;


@Slf4j
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
@Table(name = "product")
public class Product {

    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Long id;

    @Basic(optional = false)
    private String name;

    private Integer price;

    @Lob
    private String description;

    private Long stock;

    public Product(String name, Integer price, String description, UUID productCode) {
        this.name = name;
        this.price = price;
        this.description = description;
    }

    public void minusStock() {
        if (stock <= 0) {
            throw new IllegalStateException("요청하신 상품의 재고가 소진되었습니다.");
        }
        stock -= 1;
        log.info("재고: {}", stock);
    }
}

File: flyway/src/main/kotlin/study/flyway/domain/product/domain/Product.kt
package study.flyway.domain.product.domain

import javax.persistence.*

@Table(name = "product")
@Entity
class Product(
    id: Long = 0,
    name: String,
    price: Long,
    stock: Long = 0,
    description: String?,
    category: String?,
    image: String?
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id = id

    @Column(name = "name", nullable = false, length = 100)
    val name = name

    @Column(name = "price", nullable = false)
    val price = price

    @Column(name = "stock", nullable = false)
    val stock = stock

    @Column(name = "description", length = 1000)
    val description = description

    @Column(name = "category", length = 100)
    val category = category

    @Column(name = "image", length = 200)
    val image = image
}